compiler/testData/diagnostics/tests/controlFlowAnalysis/propertiesInitWithOtherInstance.fir.kt

```diff
@@ -2,11 +2,11 @@ class A(val next: A? = null) {
     val x: String
     init {
         next?.x = "a"
-        x = "b"
-        this.x = "c"
-        x = "d" // don't repeat the same diagnostic again with this receiver
-        this.x = "e"
+        <!VAL_REASSIGNMENT!>x<!> = "b"
```

Comment:
This is a false alarm, and the reason behind this is that property initialization info is kind of aggregated to property symbol only, i.e., property accesses/writes with different receivers aren't differentiated yet. I left a TODO above in a related place. A quick trial reveals this is not a trivial issue. I'll keep working on it, while these changes are reviewed. Seems not harmful for full pipeline at the moment.